corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -1,38 +1,77 @@
+            var self = {};
+            self.name = ko.observable(name);
+            self.basicTabValid = ko.computed(function () {
+                return !_.isEmpty(self.name().trim());
+            });
+            self.setRuleTabVisibility = function () {
+                if (self.basicTabValid()) {
+                    $("#rule-nav").removeClass("hidden");
+                }
+            };
+            self.navigateToNav = function (navId) {
+                $(navId).find("a").trigger("click");
+            };
+            self.handleBasicNavContinue = function () {
                 $("#rule-nav").removeClass("hidden");
-            }
-        };
-        self.navigateToNav = function (navId) {
-            $(navId).find('a').trigger('click');
-        };
-        self.handleBasicNavContinue = function () {
-            $("#rule-nav").removeClass("hidden");
-            $('#rule-nav').find('a').trigger('click');
-        };
-        return self;
-    }
+                $("#rule-nav").find("a").trigger("click");
+            };
+            return self;
+        }
+
+        function getFormContents() {
+            return _.object(
+                _.map($(".main-form").serializeArray(), function (obj) {
+                    return [obj.name, obj.value];
+                })
+            );
+        }
+
+        $(function () {
+            var name = initialPageData.get("rule_name");
+            var basicInformation = basicInformationTab(name);
+            // setup tab
+            basicInformation.setRuleTabVisibility();
+            $("#conditional-alert-basic-info-panel").koApplyBindings(
+                basicInformation
+            );
 
-    $(function () {
-        var name = initialPageData.get('rule_name');
-        var basicInformation = basicInformationTab(name);
-        // setup tab
-        basicInformation.setRuleTabVisibility();
-        $("#conditional-alert-basic-info-panel").koApplyBindings(basicInformation);
-    });
-});
+            var initialFormContent = getFormContents();
+
+            $(".main-form").on("submit", function () {
+                var finalFormContent = getFormContents();
+                // compare changes with initialFormContent to check if only 'message' has changed
+                var formContentChanges = [];
+                Object.keys(finalFormContent).forEach(function (key) {
+                    finalFormContent[key] !== initialFormContent[key]
+                        ? formContentChanges.push(key)
+                        : null;
+                });
+                var runMessagingRule = !_.isEqual(
+                    ["message"],
+                    formContentChanges
+                );
+                // to convey to HQ to reprocess or not
+                var input = $("<input />")
+                    .attr("type", "hidden")
+                    .attr("name", "runMessagingRule")
+                    .attr("value", runMessagingRule);
+                $(this).append(input);
```

Comment:
can we add this in the actual django form instead and just set it to False in case the changes are just for "message"?
I say so because this is injecting some magic via javascript which i think can be difficult for devs to notice. With a hidden input field in django form and a help text or a comment in code, would make it explicit on why that exists.